### ISSUE TITLE
Add a button for Deployment/Statefulset logs

### ISSUE THREAD

**Issue by petar-cvit** [cyclops-ui/cyclops#257]
**Is your feature request related to a problem? Please describe.**
We recently implemented an endpoint for fetching aggregated logs for Deployments and Statefulsets. It would be useful to add a button for viewing logs on Deployments and Statefulsets.

[Here](https://github.com/cyclops-ui/cyclops/blob/543d51f6435b6b2dbed2061f759157f8a8cc17a0/cyclops-ui/src/components/k8s-resources/Deployment.tsx#L346) is the modal for viewing logs on Deployment logs. The scope of the issue is to add a button on the Deployment level that would fetch logs from `/resources/deployments/:namespace/:deployment/:container/logs`

**Comment by yuvarajrece** [comment]
@petar-cvit review my PR

**Comment by petar-cvit** [comment]
Hey @yuvarajrece. Can you link the PR and the issue by commenting `closes #257` to it? Also, can you rename the PR to be more descriptive?

**Comment by doncicuto** [comment]
Hi @petar-cvit, would you assign this issue to me? Thanks!

**Comment by petar-cvit** [comment]
Sure @doncicuto

**Comment by doncicuto** [comment]
Hi @petar-cvit, I've some doubts about the current backend implementation for Deployment/Statefulset logs

I assume that you want Cyclops UI to show all the logs for every single container for every pod found in the deployment/statefulset. This would be the screenshot for the button to check that the UI would be fine:

![image](https://github.com/cyclops-ui/cyclops/assets/30386061/a41d9699-239a-404c-9231-021f4c45a860)

Right now I've readded the following endpoints that were removed by @hanshal101 (accidentally in a subsequent PR) :

```
h.router.GET("/resources/deployments/:namespace/:deployment/:container/logs", modulesController.GetDeploymentLogs)
h.router.GET("/resources/statefulsets/:namespace/:name/:container/logs", modulesController.GetStatefulSetsLogs)
```

The thing is that the endpoint requires to pass the name of the container that you want to get the logs. I've no problem to use the first container for the first pod in the deployment, but you'd miss the rest of the logs.

In summary, I'd like to know if you're ok with getting only the logs of the first container for the first pod in the deployment or you want to show all the logs something like https://github.com/stern/stern provides. 

If you prefer the second then the backend should be modified in order to get logs for all the containers and pods and therefore the container name would not be passed to the endpoint. In this case maybe it'd be better to create a new issue and either @hanshal101 or me would make the needed changes to the backend.

Just let me know which path should I follow. Thanks!

**Comment by petar-cvit** [comment]
I was thinking of having the same approach as with pod logs. Pod logs show different containers in different tabs. Could we do that?
<img width="1129" alt="Screenshot 2024-06-17 at 14 03 56" src="https://github.com/cyclops-ui/cyclops/assets/72022639/4f0fad61-6d73-47fd-a052-e71dd1fb58d0">

But yeah, you should add these back
```
h.router.GET("/resources/deployments/:namespace/:deployment/:container/logs", modulesController.GetDeploymentLogs)
h.router.GET("/resources/statefulsets/:namespace/:name/:container/logs", modulesController.GetStatefulSetsLogs)
```

**Comment by doncicuto** [comment]
Thanks @petar-cvit, I'm preparing the changes for the PR right now

**Comment by kom-senapati** [comment]
I would like to work on this @petar-cvit

**Comment by kom-senapati** [comment]
@quest-bot embark

**Comment by Sheikh-Abubaker** [comment]
@quest-bot embark
